src/journal/journald-kmsg.c

```diff
@@ -385,6 +385,40 @@ static int dispatch_dev_kmsg(sd_event_source *es, int fd, uint32_t revents, void
         return server_read_dev_kmsg(s);
 }
 
+int server_reload_dev_kmsg(Server *s) {
+        int r;
+
+        assert(s);
+
+        mode_t mode = O_CLOEXEC|O_NONBLOCK|O_NOCTTY|
+                (s->read_kmsg ? O_RDWR : O_WRONLY);
+
+        int flags = fcntl(s->dev_kmsg_fd, F_GETFL);
+        if (flags == -1) {
+                // TODO: Proceed with reload?
+                /* Since we are not sure whether mode was updated post-reload, log warning and proceed with reload. */
+                log_full_errno(LOG_WARNING, errno, "Failed to check what mode dev_kmsg_fd was opened in: %m.\n");
```

Comment:
it's fine to proceed if this fails and just close/reopen the thing.

but please use log_warning_errno() rather than log_full_errno(LOG_WARNING, …

also, do not end log messages in \n, that's appended implicitly by log_xyz() family of calls.

finally, if we ignore an error or handle it gracefully, always end the log message in ", ignoring" or so. i.e. here:

```c
log_warning_errno(errno, "Failed to check what mode kmsg fd was opened in, ignoring: %m");
```